Exercise-02/Exercise02/main.swift

```
@@ -7,63 +7,386 @@
     switch command {
-
     case "Q":
-      exit(0)
-    //TODO: implements your functions and the handling of the commands.
+        exit(0)
+    case "K", "C", "F":
+        insertTemperature(command: command, records: &records)
+    case "D":
+        deleteTemperature(records: &records)
+    case "K print", "C print", "F print":
+        printRecords(records: records, unit: String(command.prefix(1)))
+    case "print Max":
+        printMinMax(records: records, printMin: false)
+    case "print Min":
+        printMinMax(records: records, printMin: true)
+    case "average in C", "average in K", "average in F":
+        printAverage(records: records, unit: String(command.suffix(1)))
     default:
-      printInvalidCommand()
+        printInvalidCommand()
     }
   }
 }
+/// Takes in the temperature and the previous recordings and returns if the temperature is valid or not
+func tempIsValid(previousRecords: [(temperature: Float, hour: Int, day: Int)], currentRecord: (temperature: Float, hour: Int, day: Int)) -> Bool{
+    guard abs(currentRecord.temperature) < 50 else {
+        print("Temperature out of range")
+        return false
+    }
+    guard let latestRecord = previousRecords.last else {
+        // if no record is present then the temperature can be inserted
+        return true
+    }
+    if ((currentRecord.day - latestRecord.day) * 24) + (currentRecord.hour - latestRecord.hour) >= 4 {
+        return true
+    }
+    print("Temperature recorded too soon")
+    return false
+}
 
-// TODO: Add your functions
+/// Converts temperature to degree celsius
+/// - Parameters:
+///   - temperature: termperature to be converted
+///   - unit: unit of input temperature
+func temperatureToCelsius (temperature: Float, unit: String) -> Float? {
+    switch unit {
+    case "K":
+        return temperature - 273.16
+    case "F":
+        return (temperature - 32) / 1.8
+    case "C":
+        return temperature
+    default:
+        return nil
```

In a real case, here I would probably throw an error instead of using a `nil`.
Let's discuss this